app/templates/ui/app/detail/detail.controller.js

```diff
@@ -1,14 +1,49 @@
-(function () {
+(function() {
   'use strict';
 
-  angular.module('app.detail')
-    .controller('DetailCtrl', DetailCtrl);
+  angular.module('app.detail').factory('detailmodel',
+      [ 'MLRest', '$stateParams', function(mlRest, $stateParams) {
+        var uri = $stateParams.uri;
+        var model = {
+          detail : {},
+          tabs : [ {
```

Comment:
Is there a particular benefit of using an Array for the tab data? Wouldn't a hash object be nicer?